extensions/amp-playbuzz/0.1/amp-playbuzz.js

```diff
@@ -0,0 +1,263 @@
+    super(element);
+
+    /** @private {?Element} */
+    this.iframe_ = null;
+
+    /** @private {?Promise} */
+    this.iframePromise_ = null;
+
+    /** @private {?string} */
+    this.item_ = '';
+
+     /** @private {?number} */
+    this.itemHeight_ = 300; //default
+
+     /** @private {?boolean} */
+    this.displayItemInfo_ = false;
+
+     /** @private {?boolean} */
+    this.displayShareBar_ = false;
+
+     /** @private {?boolean} */
+    this.displayComments_ = false;
+  }
+  /**
+   * @override
+   */
+  preconnectCallback() {
+    this.preconnect.preload(this.item_);
+  }
+
+  /** @override */
+  renderOutsideViewport() {
+    return false;
+  }
+
+  /** @override */
+  buildCallback() {
+    // EXPERIMENT
+    AMP.toggleExperiment(EXPERIMENT, true); //for dev
+    user().assert(isExperimentOn(this.win, EXPERIMENT),
+      `Enable ${EXPERIMENT} experiment`);
+
+    const e = this.element;
+    this.item_ = user().assert(
+      e.getAttribute('item-url'),
+      'The item attribute is required for <amp-playbuzz> %s',
+      e);
+    const parsedHeight = parseInt(e.getAttribute('height'), 10);
+    this.itemHeight_ = isNaN(parsedHeight) ? this.itemHeight_ : parsedHeight;
+    this.displayItemInfo_ = e.getAttribute('display-item-info') === 'true';
+    this.displayShareBar_ = e.getAttribute('display-share-buttons') === 'true';
+    this.displayComments_ = e.getAttribute('display-comments') === 'true';
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return layout === Layout.RESPONSIVE;
+    // return layout === Layout.CONTAINER;
+    // return isLayoutSizeDefined(layout);
+  }
+
+  /** @override */
+  createPlaceholderCallback() {
+    const placeholder = this.win.document.createElement('div');
+    setStyles(placeholder, {
+      'max-height': '300px',
+      'height': '300px',
+    });
+    placeholder.setAttribute('placeholder', '');
+    placeholder.appendChild(this.createPlaybuzzLoader_());
+    return placeholder;
+  }
+
+  /** @override */
+  layoutCallback() {
+    const iframe = this.element.ownerDocument.createElement('iframe');
+    this.iframe_ = iframe;
+    iframe.setAttribute('frameborder', '0');
+    iframe.setAttribute('allowtransparency', 'true');
+    iframe.src = this.generateEmbedSourceUrl_();
+
+    this.listenToPlaybuzzItemMessage_('resize_height',
+      this.setElementSize_.bind(this));
+
+    whenDocumentComplete(this.win.document)
+      .then(this.setElementSize_.bind(this));
+
+    this.applyFillContent(iframe);
+    this.element.appendChild(iframe);
+    setStyles(iframe, {
+      'opacity': 0,
+      'min-height': '300px',
+      'height': '300px',
+    });
+    setStyles(this.element, {'height': '300px', 'min-height': '300px'});
+
+    return this.iframePromise_ = this.loadPromise(iframe).then(() => {
+      this.togglePlaceholder(false);
+      this.getVsync().mutate(() => {
+        setStyles(iframe, {'opacity': 1});
+      });
+      this.getViewport().onChanged(
```

Comment:
Can we put this in a separate PR - this will have to go through some privacy review. You want to get the data that intrinsically belongs to the embedding document. You, of course, have the ability to message yourself when the embed comes into the viewport and out of viewport - that's very typical and not controversial. Also, the page-wide analytics calls are intended to go through AMP Analytics (embed analytics are up to you). There'll be some discussion involved here and thus I recommend dropping it from this PR and doing it separately.